src/main/java/net/dv8tion/jda/api/requests/restaction/GuildScheduledEventAction.java

```diff
@@ -0,0 +1,280 @@
+ *    .setLocation("Mike's Backyard")
+ *    .queue();
+ * }</pre>
+ *
+ * @see    net.dv8tion.jda.api.entities.Guild
+ * @see    Guild#createScheduledEvent()
+ */
+public interface GuildScheduledEventAction extends AuditableRestAction<GuildScheduledEvent>
+{
+    @Nonnull
+    @Override
+    GuildScheduledEventAction setCheck(@Nullable BooleanSupplier checks);
+
+    @Nonnull
+    @Override
+    GuildScheduledEventAction timeout(long timeout, @Nonnull TimeUnit unit);
+
+    @Nonnull
+    @Override
+    GuildScheduledEventAction deadline(long timestamp);
+
+    /**
+     * The guild to create the {@link GuildScheduledEvent} in
+     *
+     * @return The guild
+     */
+    @Nonnull
+    Guild getGuild();
+
+    /**
+     * Sets the name for the new {@link GuildScheduledEvent GuildScheduledEvent}.
+     * <br><b>Note:</b> A name is required to be set before creating an event.
+     *
+     * @param  name
+     *         The name for the new {@link GuildScheduledEvent GuildScheduledEvent}
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the new name is blank, empty, {@code null}, or contains more than {@value GuildScheduledEvent#MAX_NAME_LENGTH}
+     *         characters
+     *
+     * @return The current GuildScheduledEventAction, for chaining convenience
+     */
+    @Nonnull
+    @CheckReturnValue
+    GuildScheduledEventAction setName(@Nonnull String name);
+
+    /**
+     * Sets the description for the new {@link GuildScheduledEvent GuildScheduledEvent}.
+     * This field may include markdown.
+     *
+     * @param  description
+     *         The description for the new {@link GuildScheduledEvent GuildScheduledEvent},
+     *         or {@code null} for no description.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the new description contains more than {@value GuildScheduledEvent#MAX_DESCRIPTION_LENGTH}
+     *         characters.
+     *
+     * @return The current GuildScheduledEventAction, for chaining convenience
+     */
+    @Nonnull
+    @CheckReturnValue
+    GuildScheduledEventAction setDescription(@Nullable String description);
+
+    /**
+     * Sets the cover image for the new {@link GuildScheduledEvent GuildScheduledEvent}.
+     *
+     * @param  icon
+     *         The cover image for the new {@link GuildScheduledEvent GuildScheduledEvent},
+     *         or {@code null} for no cover image.
+     *
+     * @return The current GuildScheduledEventAction, for chaining convenience
+     */
+    @Nonnull
+    @CheckReturnValue
+    GuildScheduledEventAction setImage(@Nullable Icon icon);
+
+    /**
+     * Sets the location for the new {@link GuildScheduledEvent} to take place in a specified stage channel.
+     * This will also change the event's type to {@link GuildScheduledEvent.Type#STAGE_INSTANCE Type#STAGE_INSTANCE}, and any previously set external location or
```

Comment:
```suggestion
     * This will also change the event's type to {@link GuildScheduledEvent.Type#STAGE_INSTANCE STAGE_INSTANCE}, and any previously set external location or
```